src/api/buddies.ts

```diff
@@ -17,21 +17,45 @@ const buddyType = t.intersection([
   t.partial({ status: t.literal('banned') }),
 ]);
 
-export type BanActions = 'Ban' | 'Unban';
-export type BanStatuses = 'Banned' | 'NotBanned';
+const buddiesType = t.strict({ resources: t.array(buddyType) });
+
+export type BanActions = 'Ban' | 'Unban' | 'Delete';
+export type BanStatuses = 'Banned' | 'NotBanned' | 'Deleted';
+export type BanStatusStrings = 'banned' | 'ok' | 'deleted';
 
 export type Buddy = {
   buddyId: string;
   name: string;
   status: BanStatuses;
 };
 
+export type Buddies = Record<string, Buddy>;
+
 const toBuddy = ({ id, display_name, status }: ApiBuddy): Buddy => ({
   buddyId: id,
   name: display_name,
   status: status === 'banned' ? 'Banned' : 'NotBanned',
 });
 
+const toBuddies = ({ resources }: t.TypeOf<typeof buddiesType>) =>
+  resources.reduce((acc: Buddies, apiBuddy) => {
+    const buddy = toBuddy(apiBuddy);
+
+    return { ...acc, [buddy.buddyId]: buddy };
+  }, {});
+
+export function banActionToStatusString(action: BanActions): BanStatusStrings {
+  let statusStr: BanStatusStrings = 'ok';
+
+  if (action === 'Ban') {
+    statusStr = 'banned';
+  } else if (action === 'Delete') {
+    statusStr = 'deleted';
+  }
+
+  return statusStr;
+}
+
```

Comment:
Instead of this mapping function, could we have just a json object  ==>
```
type MappedStatuses = {
    [K in BanActions]: BanStatusStrings
}

const mappedStatuses: MappedStatuses = {
    Ban: 'banned',
    Delete: 'deleted'
    Unban: 'ok'
}
```